docs/references/api/resource_embedding.rst

```diff
@@ -1196,6 +1206,40 @@ You can use this to get the columns of a join table in a many-to-many relationsh
      }
    ]
 
-.. note::
+.. _spread_to_many_embed:
 
-  The spread operator ``...`` is borrowed from the Javascript `spread syntax <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax>`_.
+Spread One-To-Many and Many-To-Many relationships
+-------------------------------------------------
+
+For these relationships, the spread embedded resource columns will be aggregated into arrays.
+
+.. code-block:: bash
+
+   # curl -g "http://localhost:3000/directors?select=first_name,...films(film_titles:title,film_years:year)&first_name=like.Quentin*"
+
+  curl --get "http://localhost:3000/directors" \
+    -d "select=first_name,...films(film_titles:title,film_years:year)" \
+    -d "first_name=like.Quentin*"
+
+.. code-block:: json
+
+   [
+     {
+       "first_name": "Quentin",
+       "film_titles": [
+         "Pulp Fiction",
+         "Reservoir Dogs"
+       ],
+       "film_years": [
+         1994,
+         1992
+       ]
+     }
+   ]
+
+The order of the values inside the resulting array is unspecified.
+However, if more than one embedded column is selected, `it is safe to assume <https://www.postgresql.org/message-id/15950.1491843689%40sss.pgh.pa.us>`_ that all of them will return the values in the same unspecified order.
+From the previous example, we can say that "Pulp Fiction" was made in 1994 and "Reservoir Dogs" in 1992.
+
+It is expected to get ``null`` values in the resulting array.
+You can exclude them with :ref:`stripped_nulls`.
```

Comment:
We had the discussion about it here: https://github.com/PostgREST/postgrest/pull/3640#discussion_r1786947751

> This will lead to inconsistent results [...] Some of them will be stripped, but others won't.

From the convo above, I was also on the fence about it, but I figured that adding "this only works on to-many spreads" to the docs would clarify some things  (I forgot to do that btw). Still, I agree, I think the inconsistency you mention is enough to look for an alternative (maybe another parameter in the header?).

Reply:
> And also any other filter on the embedding?

Yes, filters on the spread embed resource work, there's a couple of tests with them.

> Is something like this supposed to work?

Yes, it will work on a single embed resource and won't include the `null` values. But deeply nested resources could include nulls inside the array when the value is `null` or when no embedded row is returned. This is a problem with the current implementation.

Hmm... with the fixed implementation (non-flattened arrays) this *may* not be a problem anymore, since it should return empty arrays instead of `null`... but I'm not entirely sure, I need to check the new design of the resulting queries to verify.

Reply:
> Hmm... with the fixed implementation (non-flattened arrays) this may not be a problem anymore, since it should return empty arrays instead of null... 

Yes, AFAICT this is correct. Since the `array_agg` is done in the same sub-query selection, it would return `null` on a failed JOIN. But it returns `[null]` when the JOIN is successful and the value is `null` (which is what we want). So just the explicit filter should be needed here, not the header.